### ISSUE TITLE
python-2.7 ImportError: No module named request

### ISSUE THREAD

**Issue by AK-ayush** [jupyterlab/jupyterlab#4174]
In my virtual environment, python-2.7 is not able to find the module `urllib.request `and `urllib.error` required by `commands.py` in `jupyterlab` module.
 
`from urllib.request import Request, urlopen, urljoin, quote`
`from urllib.error import URLError`

`urllib.request` is now deprecated from `urllib` for python 2
@vidartf , I think you have made these changes.

**Comment by vidartf** [comment]
Ref. #3608.

**Comment by vidartf** [comment]
Anybody have an idea why CI did not pick this up?

**Comment by vidartf** [comment]
From [relevant python 2 log on #3608 build](https://travis-ci.org/jupyterlab/jupyterlab/jobs/350368318):
```
+[[ py2 == py2 ]]
+py.test -v
============================= test session starts ==============================
platform linux -- Python 3.6.4, pytest-3.4.2, py-1.5.2, pluggy-0.6.0 -- /home/travis/miniconda/envs/test/bin/python
```

**Comment by glycerine** [comment]
I get this same error when trying to run from tip, 38d7a2c0f08ea7deb40905c4d69c12b19222bb1f

osx 10.12.5, node v9.9.0, 
~~~
jaten@Jasons-MacBook-Pro ~/go/src/github.com/glycerine/jupyterlab (master) $ jupyter lab 
jupyter lab 
Traceback (most recent call last):
  File "/usr/local/anaconda2/bin/jupyter-lab", line 11, in <module>
    load_entry_point('jupyterlab', 'console_scripts', 'jupyter-lab')()
  File "/usr/local/anaconda2/lib/python2.7/site-packages/pkg_resources/__init__.py", line 572, in load_entry_point
    return get_distribution(dist).load_entry_point(group, name)
  File "/usr/local/anaconda2/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2755, in load_entry_point
    return ep.load()
  File "/usr/local/anaconda2/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2408, in load
    return self.resolve()
  File "/usr/local/anaconda2/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2414, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "/Users/jaten/dev/cnet/go/src/github.com/glycerine/jupyterlab/jupyterlab/labapp.py", line 14, in <module>
    from .commands import (
  File "/Users/jaten/dev/cnet/go/src/github.com/glycerine/jupyterlab/jupyterlab/commands.py", line 22, in <module>
    from urllib.request import Request, urlopen, urljoin, quote
ImportError: No module named request
jaten@Jasons-MacBook-Pro ~/go/src/github.com/glycerine/jupyterlab (master) $ git log
git log
WARNING: terminal is not fully functional
-  (press RETURN) 
commit 38d7a2c0f08ea7deb40905c4d69c12b19222bb1f (HEAD -> master, origin/master, origin/HEAD)
Merge: 51a26344d 4bcc9ed11
Author: Afshin Darian <git@darian.af>
Date:   Fri Mar 30 09:35:42 2018 +0100

    Merge pull request #4274 from ian-r-rose/fix_listing_shutdown
    
    Don't modify the sessions while we are iterating over them.

~~~

**Comment by glycerine** [comment]
Looks like this is due to the imports of `urllib.request` and `urllib.parse` added in https://github.com/jupyterlab/jupyterlab/pull/3608 being python3 only.

https://docs.python.org/2.7/library/urllib.html says 
~~~
The urllib module has been split into parts and renamed in Python 3 to urllib.request, urllib.parse, and urllib.error.
~~~

**Comment by glycerine** [comment]
as @vidartf pointed out above, the CI test
https://github.com/jupyterlab/jupyterlab/pull/3608 says it tested python3.6.4, not 2.7.

~~~
============================= test session starts ==============================
platform linux -- Python 3.6.4, pytest-3.4.2, py-1.5.2, pluggy-0.6.0 -- /home/travis/miniconda/envs/test/bin/python
cachedir: .pytest_cache
rootdir: /home/travis/build/jupyterlab/jupyterlab, inifile: pytest.ini
plugins: check-links-0.2.0
collected 31 items                                                             
~~~

**Comment by jasongrout** [comment]
> Anybody have an idea why CI did not pick this up?

It looks like there is shortcut if the miniconda directory already exists to skip the installation. My guess is that the miniconda directory was cached for a python3 install, but then that skipped the python 2 install. I'm working on a PR that uses different miniconda directories for python2/python3 to avoid this, and make sure we are getting the right miniconda to run the test.

**Comment by jasongrout** [comment]
cf. https://github.com/jupyterlab/jupyterlab/blob/4e0e2580c20a5d4947f7e654478605d54750b6ec/scripts/travis_install.sh#L8-L9

**Comment by jasongrout** [comment]
Turns out that was a red herring. Instead, we should ensure that python is installed in the test conda environment. #4345 fixes this.

**Comment by vidartf** [comment]
This should be fixed by #4289 and #4345.
